crates/spidermonkey-embedding-splicer/src/splice.rs

```diff
@@ -757,32 +757,50 @@ fn synthesize_import_functions(
             .get_fn_modifier(coreabi_get_import_fid)
             .unwrap();
 
-        // walk until we get to the const representing the table index
-        let mut table_instr_idx = 0;
-        for (idx, instr) in builder.body.instructions.iter_mut().enumerate() {
-            if let Operator::I32Const { value: ref mut v } = instr.op {
-                // we specifically need the const "around" 3393
-                // which is the coreabi_sample_i32 table offset
-                if *v < 1000 || *v > 5000 {
-                    continue;
+        // Find the I32Const base index and compute the delta to new base
+        let mut table_instr_idx = 0usize;
+        let mut delta: i32 = 0;
+        {
+            let ops_ro = builder.body.instructions.get_ops();
+            for (idx, op) in ops_ro.iter().enumerate() {
+                if let Operator::I32Const { value } = op {
+                    // we specifically need the const "around" 3393
+                    // which is the coreabi_sample_i32 table offset
+                    if *value < 1000 || *value > 5000 {
+                        continue;
+                    }
+                    delta = import_fn_table_start_idx - *value;
+                    table_instr_idx = idx;
+                    break;
                 }
-                *v = import_fn_table_start_idx;
-                table_instr_idx = idx;
-                break;
             }
         }
-        builder.inject_at(
+        // Inject adjustments after the located instruction: add delta and add arg index
```

Comment:
Why doesn't the `inject_at` approach work here anymore? Maybe you just couldn't find it anymore because it moved into the `InjectAt` trait? If you change the `use wirm::opcode::Inject;` statement to `use wirm::opcode::{Inject, InjectAt};` I think the original code should work as before.

Reply:
Yeah I totally missed the `InjectAt`, added it back but the other code is necessary to set the instruction as it can't be done in the loop for the reason in the previous comment